activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -53,7 +64,14 @@ def self.initialize_i18n(app)
       # Restore available locales check so it will take place from now on.
       I18n.enforce_available_locales = enforce_available_locales
 
-      reloader = ActiveSupport::FileUpdateChecker.new(I18n.load_path.dup){ I18n.reload! }
+      directories = Helpers.watched_dirs_with_extensions(reloadable_paths)
+      reloader = ActiveSupport::FileUpdateChecker.new(I18n.load_path.dup, directories) do
+        I18n.load_path.keep_if { |p| File.exist?(p) }
+        I18n.load_path |= reloadable_paths.map(&:existent).flatten
```

Comment:
`reloadable_paths` is initially `nil`, is it guaranteed to have reached [here](https://github.com/rails/rails/pull/21124/files#diff-5a01fa54cd7e307fec45a4c5d819c63aR53) when this line is run?